src/vma/sock/sock-redirect.cpp

```diff
@@ -1626,13 +1630,15 @@ inline int epoll_wait_helper(int __epfd, struct epoll_event *__events, int __max
 		epoll_wait_call epcall(extra_events_buffer, NULL,
 				__epfd, __events, __maxevents, __timeout, __sigmask);
 
-		int nfds = epcall.get_current_events();
-		if (nfds > 0) {
-			return nfds;
+		int rc = epcall.get_current_events(); // returns ready nfds
+		if (rc <= 0) {
+			// if not ready nfds then check all lower level queues (VMA ring's and OS queues)
+			epcall.init_offloaded_fds();
+			rc = epcall.call();
 		}
 
-		epcall.init_offloaded_fds();
-		return epcall.call();
+		srdr_logfunc_exit("rc = %d", rc);
```

Comment:
I prefer to leave as rc which justifies the general case.
Since the epoll_wait() man pages explain the 3 options possible for this return code. Only the 'good' one is nfds, the rest is timeout (rc=0) or error (rc=-1 + errno)